models/concerns/expand_data.rb

```diff
@@ -41,10 +41,13 @@ def add_sticky_vehicle_if_routes_and_partitions
   end
 
   def clean_according_to(unfeasible_services)
-    unfeasible_services.each{ |unfeasible_service|
-      self.routes.each{ |route|
-        route.mission_ids.delete_if{ |id| id == unfeasible_service[:original_service_id] }
-      }
+    unfeasible_services_ids = unfeasible_services.collect{ |un| un[:original_service_id] }
+    self.routes.each{ |route|
+      route.mission_ids.delete_if{ |id| unfeasible_services_ids.include?(id) }
+    }
+
+    self.relations.each{ |relation|
+      relation.linked_services.delete_if{ |service| unfeasible_services_ids.include?(service.id) }
```

Comment:
Right, I remember having a bug because of the fact that some relations where pointing at services that do not exist anymore in vrp (because they were rejected by detect_unfeasible) and I did not realize this is already taken into account. 

I can indeed skip these lines.

Reply:
In theory, we should clean all non existent objects but I don't think `clean_according_to` is the right way to do it. So yes you can remove the route cleaning if ortools has the same protections wrt invalid ids.

Reply:
I think the right way to do all this cleaning is "add_unassigned" (even the `vrp.services -= unasinged` logic should be there) -- that is, the meaning of "filter infeasible" should include all the related logic of removing a service...